include/mma_tensor_op/gemm.cuh

```diff
@@ -0,0 +1,547 @@
+#pragma once
```

Comment:
doxygen---I don't see anyone else digging into the weeds of this code, but a broad understanding of what each function does would help the code review and be good for future reference

Reply:
Sorry about the lack of docs here: will add as soon as I can.